rewrite-gradle/src/main/resources/META-INF/rewrite/gradle-6.yml

```
@@ -0,0 +1,51 @@
+#
+# Copyright 2024 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.gradle.MigrateToGradle6
+displayName: Migrate to Gradle 6 from Gradle 5
+description: Migrate to version 6.x. See the Gradle upgrade guide from [version 5.x to 6.0](https://docs.gradle.org/current/userguide/upgrading_version_5.html) for more information.
+recipeList:
+  - org.openrewrite.gradle.MigrateToGradle5
+  - org.openrewrite.gradle.UpdateGradleWrapper:
+      version: 6.x
+      addIfMissing: false
+  # We can't easily convert `compile` to either `api` or `implementation`, because transitive requirements are unclear
+  # https://github.com/openrewrite/rewrite/issues/4194
+  - org.openrewrite.gradle.ChangeDependencyConfiguration:
+      groupId: "*"
+      artifactId: "*"
+      configuration: testCompile
+      newConfiguration: testImplementation
+  - org.openrewrite.gradle.ChangeDependencyConfiguration:
+      groupId: "*"
+      artifactId: "*"
+      configuration: runtime
+      newConfiguration: runtimeOnly
```

supernit: In practice this doesn't change the meaning, but generally when I'm thinking about these sort of changes, I'm thinking in terms of `main` and then `test`.

```suggestion
  - org.openrewrite.gradle.ChangeDependencyConfiguration:
      groupId: "*"
      artifactId: "*"
      configuration: runtime
      newConfiguration: runtimeOnly
  - org.openrewrite.gradle.ChangeDependencyConfiguration:
      groupId: "*"
      artifactId: "*"
      configuration: testCompile
      newConfiguration: testImplementation
```